### DISCUSSION TITLE
Possible separation of MME and S/PGW services of AGW

### DISCUSSION THREAD

**Discussion by MAAHamid** [magma/magma#12208] [Q&A] [answered]
Hello,

Magma AGW services weren't designed such that it allows the separation between MME or S/PGW services and it is treated as a whole EPC in a box. 
For some of the use cases it is benefecial to separate MME, SGW and PGW (for example; proper scalling up or bumping the user plane without the necessity of adding resources to the control plane, other example, deploying the S/PGW (UP) on the far edge while keeping the MME (CP) centralized on certain core network).

Discussion questions,
- Is this somehow possible with current AGW design?
- Will this be possible after implementing the AGW as a CNF?

I know mostly such separation isn't possible, let me know your ideas or advices.

**Comment by ssanadhya** [comment]
@MAAHamid , you can build the MME service of AGW with [`FEATURES=mme_oai`](https://github.com/magma/magma/blob/master/lte/gateway/Makefile#L9-L13) to have a standalone MME with standard S11 interface over GTPv2c. Team at OAI uses this with their own CUPS SPGW. cc: @lionelgo

**Reply by MAAHamid** [reply]
Thanks for the answer, may I have more info about this solution?
Will the MME work in 1:1 manner or 1:multi? Is it fully standalone or coupled with SPGW services

**Reply by lionelgo** [reply]
Only 1:1 is possible, it is not coupled with SPGW services.

**Comment by ulaskozat** [comment, marked as answer]
@MAAHamid in Magma, MME and part of the SPGW-C run as part of the mme service, session management runs as part of sessiond and data path is the pipelined service. One should not forget about Magma sctpd service which also in a traditional network is part of the MME service. Sctpd service uses unix socket to communicate with mme service (using grpc), while the other services utilizes network sockets (using grpc).  So in theory, if you package sctpd and mme in one pod, while placing other services in their corresponding pods, change service end point configurations with routable IP addresses, you can deploy them separately. Still though the relationship between these services are 1-1. To really make the relationship many to many, you need to do more changes in the source code and more importantly change some of the design patterns.

Long story short, Magma's scaling model operates at AGW level (so think AGW as your pod for replication). The overall design principles are really designed around this with Orc8r providing a single pane control/visualization of all AGWs in a network. If this design pattern is not a good fit for your use case, maybe Magma is not what you are looking for.

**Reply by MAAHamid** [reply]
Hello @ulaskozat , thanks for the detailed answer, I really appreciate it.
